mlflow/store/model_registry/file_store.py

```diff
@@ -713,6 +725,12 @@ def _get_registered_model_version_tag_path(self, name, version, tag_name):
                 f"Model Version (name={name}, version={version}) not found",
                 RESOURCE_DOES_NOT_EXIST,
             )
+        model_version = self._get_model_version_from_dir(registered_model_version_path)
+        if model_version.current_stage == STAGE_DELETED_INTERNAL:
+            raise MlflowException(
+                f"Model Version (name={name}, version={version}) not found",
```

Comment:
Would it be more helpful to the user to indicate that the registered model has been marked for deletion instead of saying that's it not found?

Reply:
It would, but I think this would require Databricks backend update.

Reply:
If we say `deleted`, users might think `can we retrieve deleted model versions?` or `can we restore them`.